tool/tsh/tsh.go

```diff
@@ -320,12 +333,47 @@ func onLogin(cf *CLIConf) {
 		utils.FatalError(trace.BadParameter("invalid identity format: %s", cf.IdentityFormat))
 	}
 
+	// Get the status of the active profile ~/.tsh/profile as well as the status
+	// of any other proxies the user is logged into.
+	profile, profiles, err := client.Status("", cf.Proxy)
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			utils.FatalError(err)
+		}
+	}
+
 	// make the teleport client and retrieve the certificate from the proxy:
 	tc, err = makeClient(cf, true)
 	if err != nil {
 		utils.FatalError(err)
 	}
 
+	// client is already logged in and profile is not expired
+	if profile != nil && !profile.IsExpired(clockwork.NewRealClock()) {
+		switch {
+		// in case if nothing is specified, print current status
+		case cf.Proxy == "" && cf.SiteName == "":
+			printProfiles(profile, profiles)
+			return
+			// in case if parameters match, print current status
+		case host(cf.Proxy) == host(profile.ProxyURL.Host) && cf.SiteName == profile.Cluster:
```

Comment:
What happens if the CA gets rotated but this certificate is still valid? If the user tries to login again, will it just keep printing the current certificate (it's valid because the TTL has not expired) but will be rejected by the cluster because the CA changed?

Reply:
the certificate will become rejected when user will try to login, so user will have to `tsh logout` and then login again.